### PR TITLE
Java: Use Java 7 try-with-resources for AutoCloseable resources #8213

### LINKED ISSUE #8213: Java: Use Java 7 try-with-resources for AutoCloseable resources
Stuffs like [this](https://github.com/TEAMMATES/teammates/blob/master/src/test/java/teammates/test/driver/FileHelper.java#L23-L32):
```java
BufferedReader br = new BufferedReader(new FileReader(filePath));
Scanner sc = new Scanner(br);
try {
    return sc.useDelimiter("\\Z").next();
} finally {
    sc.close();
    br.close();
}
```
can be modified to use the [try-with-resources statement](https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html):
```java
try (Scanner sc = new Scanner(new BufferedReader(new FileReader(filePath)))) {
    return sc.useDelimiter("\\Z").next();
}
```
The benefit is two-fold: makes the code shorter and also closes the linked resources without fail, although admittedly I cannot say for sure what will happen if the resources do leak.

There are more than 10 places in the code base where this can be applied. Some common [`AutoCloseable`](https://docs.oracle.com/javase/8/docs/api/java/lang/AutoCloseable.html) resources are `FileWriter`, `BufferedWriter`, `PrintWriter`, `FileOutputStream`, `OutputStreamWriter`, `Scanner`, `FileReader`, `BufferedReader`, `FileInputStream`, `InputStreamReader`.

### PR DESCRIPTION
**Pull request by judeibe**
Fixes #8213

### CODE REVIEW

**Comment by judeibe** [comment]
fixed commented on issues

File: src/main/java/teammates/common/util/GoogleCloudStorageHelper.java [hunk 1]
```diff
@@ -131,11 +130,12 @@ public static String getNewUploadUrl(String callbackUrl) {
         while (size > 0) {
             // Make sure not to over-read
             int bytesToRead = Math.min(size, MAX_READING_LENGTH);
-            InputStream blobStream = new BlobstoreInputStream(blobKey, offset);
-            byte[] array = new byte[bytesToRead];
-
-            blobStream.read(array);
+            byte[] array;
+            try (InputStream blobStream = new BlobstoreInputStream(blobKey, offset)) {
+                array = new byte[bytesToRead];
```
[the remark this answers is not in the document]
Comment (wkurniawan07) [reply src/main/java/teammates/common/util/GoogleCloudStorageHelper.java]: This is not addressed yet.

File: src/main/java/teammates/common/util/GoogleCloudStorageHelper.java [hunk 2]
```diff
@@ -130,12 +128,12 @@ public static String getNewUploadUrl(String callbackUrl) {
...
         while (size > 0) {
             // Make sure not to over-read
-            int bytesToRead = Math.min(size, MAX_READING_LENGTH);
-            InputStream blobStream = new BlobstoreInputStream(blobKey, offset);
-            byte[] array = new byte[bytesToRead];
-
-            blobStream.read(array);
+            byte[] array;
+            try (InputStream blobStream = new BlobstoreInputStream(blobKey, offset)) {
+                array = new byte[Math.min(size, MAX_READING_LENGTH)];
```
Comment (wkurniawan07) [inline src/main/java/teammates/common/util/GoogleCloudStorageHelper.java]: This is not what I meant. Why not just assign the value of `array` when the variable is initialized?
```java
byte[] array = ...;
try (...) {
    ...
}
```

**Comment by wkurniawan07** [comment]
@judeibe any updates here?

**Comment by wkurniawan07** [comment]
@judeibe closing due to inactivity for a long period. Feel free to request for reopen when you're ready to resume the work done here.

**Pull request closed without merging**